File: day-005/PallindromeLikedList.cpp
bool isPalindrome(Node *head)
    {
        //Your code here
          int num =0;
        for(Node*p = head ;p!=NULL; p = p->next){
            num = num*10+p->data;
        }
          int temp_num = num;
          int new_num = 0;
        while(num){
            int rem = num%10;
            num = num/10;
            new_num = new_num*10+rem;
        }
        return(new_num==temp_num)?true:false;
    }

File: day-005/PallindromeLikedList.java
public boolean isPalindrome(ListNode head) {
        Stack<Integer>st =new Stack<Integer>();
        
        int size=0;
        for(ListNode p=head; p!=null; p=p.next){
              size++;
        }
        int tempsize=size;
        size=size/2;
        ListNode p=head;
        
        for(int i=0;i<size;i++){
            st.push(p.val);
            p=p.next;
        }
        
        if(tempsize%2!=0){
            p=p.next;
        }
        for(int i=0; i<size; i++){
            int data = st.pop();
            
            if(data!=p.val)
                return false;
            p=p.next;
        }
          return true;  
        
    }
